### DISCUSSION TITLE
Wie kodiert AAX Audioconverter Umlaute bei Kapitelnamen in MP3s ?

### DISCUSSION THREAD

**Discussion by alurio** [audiamus/AaxAudioConverter#205] [Q&A]
Lieber Audiamus, 

Das folgende  hat nur indirekt mit deiner Software zu tun, aber vielleicht kannst du dennoch helfen. 

Ich  nutze zur Erstellung meiner Bücher den jedes Kapitel eine MP3 Modus und stehts auch Kapitel Namen. Die Bücher höre ich dann über  den Populären  Amdroid Hörbuch Player Smart Audiobook Player. Der sucht seit einem Update immer  danach ob sich in einer MP3 Datei mehrere Kapitel verbergen und hat dann die Kapitelnamen meiner MP3s  oft doppelt angezeigt. Inzwischen hat der Entwickler  dieses Verhalten fast ausgemerzt außer bei Kapiteln die Umlaute im Namen haben. Da ist die Software überfordert. und zeigt den Kapitelnamen einmal Korrekt und einmal verstümmelt. Hier mal der Screenshoot. 

![Screenshot_2024-02-26-21-32-25-87_c37fa21bbd5feae9cb8677086a3c7cf0](https://github.com/audiamus/AaxAudioConverter/assets/61393505/de546a4c-e0dd-47c7-bf3e-1bc810363646)

Ich  meine mich zu erinnern das es unterschiedliche Arten der Kodierung von Ä Ö Ü  gibt  die man  benutzen kann. Meine Schwache Hoffnung ist, das du und der Macher des  Smarrt Audiobook Players unterschiedliche Methoden benutzen, die man nur angleichen müsste.  Der Entwickler des Players ist gerade ratlos und der Perfektionist in mir genervt. Ich mag nicht, das jedes Kapitelname mit Umlaut falsch ausgegeben wird  :-) 

Vielleicht könntest du bei Gelegenheit mal Kontakt mit Alexander Kravchenko aufnehmen?  Seien Entwickler Mailadresse  lautet smart.abp@gmail.com

Viele Grüße Henning Alias Aurio

**Comment by audiamus** [comment]
Die Standard-Repräsentation für Text ist heutzutage Unicode, üblicherweise codiert als UTF8. AAX Audio Converter greift hier fast nicht ein, denn auch Audible verwendet UTF8. Die einzige Ausnahme in AAX Audio Converter ist die Playlist, die wahlweise als ISO Latin1 (ISO 8859-1) geschrieben werden kann. Grund für diese Wahlmöglichkeit waren bestimmte Infotainmentsysteme im Auto, die zwar eingebettetes UTF8 in den Metadaten richtig wiedergaben, aber Dateinamen mit Umlauten nicht fanden.

UTF8 ist ein 8-Bit-Zeichensatz, bei dem ein Zeichen mindestens ein Byte lang ist. Die unteren 7 bit entsprechen dem ASCII-Zeichensatz.  Umlaute und andere Zeichen außerhalb des ASCII-Umfangs werden durch Escape-Sequenzen eingeleitet. Dazu dienen die oberen 8 bit. Für ein solches Zeichen wird dann mehr als ein Byte benötigt, für deutsche Umlaute mit dem Escape 2 Byte und für bestimmte asiatische oder afrikanische Schriften sogar bis zu 4 Byte extra. Es gibt bei Unicode und UTF8 keine sprachspezifischen Varianten, deswegen auch der Name.  

ISO Latin 1 (8859-1) ist ein 8-Bit-Zeichensatz, bei dem jedes Zeichen genau 1 Byte lang ist. Die unteren 7 bit sind ASCII, mit den oberen darstellbaren 7 bit lassen sich die westeuropäischen Schriften darstellen. Die Anwendung muss wissen, dass Latin1 verwendet wird. Die Windows-Codepage 1252 entspricht weitgehend Latin1. Aber auch Windows verwendet seit Jahren Unicode. Es gibt weitere Latin/8859-Varianten, so für slawische Sprachen oder auch Latin15, wie Latin1 aber mit €-Zeichen. Wurde dann nicht mehr wirklich benötigt, weil sich Unicode durchsetzte 

Moderne Software erkennt UTF8 automatisch und auch die Codier-Reihenfolge bei Multibyte-Zeichen (Little/Big Endian). UTF8 kann aber auch einen zusätzlichen Byte-Order-Marker enthalten (BOM). Falls eine Software UTF8 nicht lesen kann, werden unsinnige Sonderzeichen angezeigt, wie in Deinem Beispiel. 

Kann es sein, dass in diesem Beispiel der obere Text aus einer ISO-Latin1-Playlist stammt, der untere aus den eingebetteten Metadaten? Oder ist das auch wieder so ein Fall wie bei den erwähnten Autoradios?

**Comment by alurio** [comment]
Lieber Audiamus 
herzlichen Dank für deine Ausführlichen Erläuterungen. 
Also an der M3U Playlist liegt es nicht. Die habe ich  gerade aus dem betroffenen Hörbuchordner gelöscht, was nichts ändert. 

Das Problem ist  wie gesagt entstanden, seit Smart Audiobook Player bei jedem Buch im MP3 Format  automatisch scannt ob eine Mp3 mehr als ein Kapitel. enthält  was bei meinen mit deiner Software erzeugten Büchern ja nie der Fall ist. Da ist eine Datei immer ein Kapitel und gut.

Der  Player zeigt bei dieser Beispieldatei halt immer entweder den Dateinamen also Vater der Lüge . MP3  oder falls man ihn auf die Nutzung von Id3 Tag schaltet nur Vater der Lüge  und darunter nochmal die Verstümmelung. wie im Bild oben.  Bis vor kurzem gab es diese Doppelung bei jedem Buch, aber inzwischen kann die Software das ausfiltern, außer eben es kommen Umlaute ins spiel., 

Nach einem längeren Mailwechsel  schrieb mir der Entwickler des  Smart Audiobook Players auf Englisch folgendes. zur Kodierung die er  für Smart Audiobook Player benutzt  

The problem with encoding still exists on my phone.
But maybe it is somehow related to my phone.
By default the app used ISO 8895-1 encoding.
But even after adding encoding auto detection the problem still persists on my phone.

Wenn  ich  dich richtig verstehe benutzt der also nicht UTF 8  , wie alle sondern Iso   Latin 1  Ist das vielleicht einfach des Rätsels Lösung müsste der einfach UTF8  für seine komische  Kapitel Detention nutzen und alles ist gut ?  Der  ist, was ja auch der Name nahe legt  mutmaßlich Osteuropäer. zumindest stand bis 2022  in Google Play als Standort Kiew.  

Wenn du Lust und Kapazität hast  würde ich mich freuen, wenn du ihm mal schreibst.. Du kannst ihn von Henning grüßen. dann weiß er schon bescheid:-)  Vielleicht dringst du noch zu ihm durch, so von Entwickler zu Entwickler.  Wäre ja auf jeden Fall  Wünschenswert wenn deine sehr populäre Software und der führende Android Hörbuchplayer wieder reibungslos zusammenarbeiten. 

 Meine Mails zu dem Problem werden inzwischen Ignoriert, ich bin ja nur User und er schreibt mir,  er hätte keine Idee wie er das Problem lösen soll.
 Viele Grüße 
 
 PS ist das Projekt AAX Audioconverter und Booklibconnect zusammenzuführen eigentlich noch aktuell?

**Comment by audiamus** [comment]
> Wenn ich dich richtig verstehe benutzt der also nicht UTF 8 , wie alle sondern Iso Latin 1 Ist das vielleicht einfach des Rätsels Lösung müsste der einfach UTF8 für seine komische Kapitel Detention nutzen und alles ist gut ?

Ggf muss er da eine zusätzliche Erkennung einbauen. Man kann mit recht einfachen Abfragen sowohl erkennen, ob es prinzipiell ISO 8859 oder UTF8 ist und dann entsprechend agieren. 

> Smart Audiobook Player

Ich benutze auf Android VLC und Voice. Gerade Voice hat sich gut entwickelt und akzeptiert inzwischen auch die AAX Audio Converter Dateistruktur mit unterteilten Kapiteln. 


> PS ist das Projekt AAX Audioconverter und Booklibconnect zusammenzuführen eigentlich noch aktuell?

Steht weiterhin auf der Liste, hat aber zugegebenermaßen nicht die höchste Priorität. Es läuft ja alles. Und ich habe diverse andere Softwareprojekte in den letzten Monaten gehabt, die in ganz anderen Bereichen spielen, überhaupt nichts mit Medien zu tun haben, aber hochspannend sind und mein Gehirnschmalz fordern.

**Comment by alurio** [comment]
Hallo Audiamus !
Okay ich lass Mal etwas Zeit vergehen und dann  Versuche ich deine Erklärungen  mit Deepl  in Englisch zu übersetzen,  und schick sie an den Player Entwickler. vielleicht wird er ja dann nochmal aktiv  und versucht das Problem zu lösen.

Viele Grüße  !
